# draft of skill
### preprocessing list of skill, remove the empty row, dulipate row 
### skill is taking from below Reference 
#### https://resumegenius.com/blog/resume-help/hard-skills
#### https://novoresume.com/career-blog/most-important-skills-to-put-on-your-resume                   

In [1]:
import pandas as pd
from nltk.stem import PorterStemmer
from nltk.tokenize import sent_tokenize, word_tokenize
ps = PorterStemmer()
skill=pd.read_csv('draftskill/skill.csv')  

In [2]:
#delete the empty row
skill.dropna(inplace=True)
#delete the repeated row
skill.drop_duplicates(keep="first",inplace=True)

In [3]:
skill

list of skill
0                         ei
1            time management
2    effective communication
3     emotional intelligence
4        conflict management
..                       ...
202             cryptography
204                      c++
205                     java
206                   jquery
207                   python

[179 rows x 1 columns]

# catorgory the draft of skill
### above contain 179 rows
### since some skills have different word length,catgory skill in different word length. 
### below code help us to get one word length skills, two words length and three words length skills

In [4]:
#create the set 
allof={"0"}
# replace the white space as ### in dataframe, and save it in the set
for i in skill['list of skill']:
    allof.add(i.replace(" ", "###"))
allof.remove("0")

In [5]:
#the set: "single" keeping one word skill
# mulltiple1 : keeping the reset, including two/three word skill
single={"1"}
multiple1={"2"}
for i in allof:
    if "###" not in i: 
        single.add(i)
    else:
        multiple1.add(i)
single.remove("1")
multiple1.remove("2")

In [6]:
# seperate two/three word skill into difference set. a contain two word skill, b contain three word skill
a={"1"}
b={"2"}
for i in multiple1:
     if(len(i)<25):# this length is based on the obseration of the data set
        a.add(i.replace("###", " "))
     else:
        b.add(i.replace("###", " "))
        
a.remove("1")
b.remove("2")

In [7]:
# some item has been catorgry by mistaking, fixing it after observation of the data
a.remove('attention to detail')
a.remove('ios app development')
b.add('attention to detail')
b.add('ios app development')
b.remove('computer troubleshooting')
b.remove('effective communication')
b.remove('networking communication')
a.add('computer troubleshooting')
a.add('effective communication')
a.add('networking communication')

In [8]:
# below len matching the length of our list of skill
len(single)+len(a)+len(b)

179

### Draft of skill, become: (single:n=1 gram).... (a:bigram)........( b: n=3 gram). 
### we will generate the n=1-3 gram of our data get from indeed. and counting term fruency base on our draft of skill
### these three kinds of grams we got it work like a fliter, we will then keep the the skill which frequency is bigger than 0 in the skill set. 
### adding some higher frequncy item from our srape data in draft skill 
### after that our draft skill might become the set of skill that require when getting a job

In [9]:
#single

In [10]:
#a

In [11]:
#b

In [12]:
s1 = list(single)
df1=pd.DataFrame(s1)

s2 = list(a)
df2=pd.DataFrame(s2)

s3 = list(b)
df3=pd.DataFrame(s3)

In [13]:
# before stemmer
df1.to_csv('skillset/skill_len_1.csv', index=False) 
df2.to_csv('skillset/skill_len_2.csv', index=False) 
df3.to_csv('skillset/skill_len_3.csv', index=False) 

In [14]:
#after stemmering

In [15]:
for i in range(0,len(s1)):
    s1[i]=ps.stem(s1[i])
    
df4=pd.DataFrame(s1)  

In [16]:
for i in range(0,len(s2)):
    s2[i]=ps.stem(s2[i])
df5=pd.DataFrame(s2)   

In [17]:
for i in range(0,len(s3)):
    s3[i]=ps.stem(s3[i])
df6=pd.DataFrame(s3)   

In [18]:
# before stemmer
df4.to_csv('skillset/ste_len_1.csv', index=False) 
df5.to_csv('skillset/ste_len_2.csv', index=False) 
df6.to_csv('skillset/ste_len_3.csv', index=False) 
print("done")

done


In [ ]:
### next move back to findskill